### DISCUSSION TITLE
Convert the Principles & Patterns Project into the Education Working Group

### DISCUSSION THREAD

**Discussion by jawache** [Green-Software-Foundation/oc#15] [General]
@tmcclell, @greenhsu123, @aecurrie 

@camcash17 is suggesting another new project to be managed by the principles & patterns team. If accepted that would make 3 projects managed by one team. Which strongly indicates Principles & Patterns isn't a project, it's a Working Group which runs 3 projects and each of those project should have it's own project leads.

- Each project needs a lead and with the current structure leading both the training and patterns projects falls under @greenhsu123 which is a lot of responsibility and time commitment, if @camcash17's idea is accepted then all 3 projects would be led by @greenhsu123 - it's not scalable.
- I propose we find separate leads for the principles, patterns and the exam project @camcash17 is proposing.
- Then I propose we form another working group, education working group, under the community working group.
- The education working group would have 3 co-chairs, I propose we form those co-chairs from the project leads of the existing projects.

We discussed this on the run up to COP but the team was super busy getting ready for that so we just pushed the conversation to later, I propose we take a look at it now.

**Comment by tmcclell** [comment]
Would the new working group also work with universities or educational institutions?

**Reply by jawache** [reply]
We'd need to define a scrope and that scope should be a subset of the scope for the community wg, but yes with that caveat in mind it could also include working with universities or educational institutions.

**Reply by Sealjay** [reply]
Definitely ok with this; got something to mention in the OC, or via email. Will include @tmcclell  @greenhsu123

**Comment by greenhsu123** [comment]
Could you share what the exam project would entail? How would it fit with Principles & Patterns? I agree that having one chair managing three projects would be too much.

**Reply by jawache** [reply]
It's not been defined in any detail, just a chat @camcash17 and I had - were waiting till mid Feb to bring it up next P&P project meeting. Similar to the idea that @aecurrie was discussing a long time ago, a much more detailed exam, you'd have to really put time in (weeks) in order to pass. We would just admin the exam, point people to existing content, other books, patterns all sorts of things, have a reading list etc... 

The kind of exam whereas if you met someone who had passed this exam you could feel confident if you hired them for your project they would be able to reduce emissions.

**Reply by greenhsu123** [reply]
Is the scope more about creating the content that goes into the exam? Or it's a process where we can feed in content from different projects across the foundation?  btw- what's a P&P project meeting?

**Reply by jawache** [reply]
Principles & Patterns meeting 😁

I believe the idea is to just create the exam, just the questions and the source material is anything, outside the Foundation would be totally fine. So more like a uni course, read all these books, papers, articles, now answer these questions. But I'm jumping the gun, it's @camcash17 idea ☺️

**Reply by camcash17** [reply]
@jawache covered it pretty well, essentially put together a new exam that requires a bit more effort to study for and understand core concepts (ie including questions to actually calculate SCI, or understanding various patterns in the catalog). As Asim mentioned, could also use source material from outside the Foundation. 

Happy to discuss this in more detail during the next call!

**Reply by greenhsu123** [reply]
The "thing" (we haven't signed the contract yet hence the "thing") that @aecurrie, @SaraEmilyBergman and I will work on for the rest of the year could go hand in hand with what we are proposing here. I think we just need to be a bit mindful of any potential repeated effort in this space. We should also take this away and check on any potential licensing issues.

**Reply by Sealjay** [reply]
There's a lot of material we can contribute - I shared a few training courses we created for L300 & hands-on materials too, with Asim prior to COP

**Comment by Sealjay** [comment]
If three projects are being managed by one team, could it also be that the projects aren't separate? I worry if everyone is a lead, then no one is doing; if I look at how the Standards WG operates, that seems pretty collegiate and like most of the work is compelted in the WG call.

**Reply by jawache** [reply]
The projects are really quite separate, as in separate repos, separate sets of issues. There is one meeting currently but it's run very differently to all our other projects as in there are two separate agendas per meeting.

E.g. this is the patterns agenda: https://github.com/Green-Software-Foundation/patterns/issues/210
and this is the principles agenda: https://github.com/Green-Software-Foundation/learn/issues/157

Both agendas are run though the same meeting and we had two co leads, Sarah drove the learn work and Bill drove the patterns work. So effectively it is two projects with two leads but we just never really acknowledged it as so. But each project does need a lead, since Bill left the patterns project hasn't had a single additional pattern added to it :/ 

From our perspective the goal is to make every WG and Project run the same way, so we have one set of docs/process/tools and rules. I'm pretty open to how we get there.

**Reply by greenhsu123** [reply]
This is similar to my worry that we are merging similar-ish projects into a sub-WG group because we don't have enough people to work on it. We are stretching our bandwidth to cover as much as possible, which may be too much of an ask in the current climate. (Looking back, we had about 2 to 3 people actively working on either Principles or Patterns, and those people aren't necessarily working on both projects) I propose we don't work on the 3rd project till at least Q4 this year (merged or not merged), giving us time to get Patterns into good shape (including validating the effectiveness and usefulness of the patterns). 

I am happy to try out the sub-working group if the rest of the project team is okay-ish with it. I just worry people will lose interest if we ask them to work on things they aren't interested in.  

One last point...one thing I see (from my very narrow view) is that we have too many projects but too few contributors. I understand the foundation is hiring PMs and developers to solve this issue, but maybe we should consider only focusing on one project at a time for the sub-WG. It had worked well so far that Principles finished before Patterns started, and both projects had enough momentum to meet their objectives.

**Reply by jawache** [reply]
Gotcha ok, so sounds like I looked at the problem the wrong way around, I assumed everyone in the P&P team was working on all projects so it was running like a pseudo WG, but sounds like it was two sperate projects run out of the same virtual meeting for convenience. Perhaps that was a good way to start this off but I think both projects have evolved to the point where certain people only want to work on one project, I've got some people who've reached out and **only** want to work on patterns for instance.

I was just tying to find a way to make this team work like the other projects in the Foundation but from what you wrote above it sounds like it could just be resolved by making sure the projects run separately. 

So I propose we just split off the projects and ask people which ones they want to work on, so the learn team has their own email list and meeting, the patterns team has their own email list and meeting (I think Markus would be a good lead for that) and if if @camcash17's project is approved then he get's his own email list, repo and meeting and if people are interested in joining him they can. 

That's how all other projects in the Foundation work and would remove this accidental implicit expectation in the P&P team that everyone works on every project which I think is causing the stress above. Apologies for that I can see how this evolved now, honestly every other project in the Foundation works via interest, if people are interested in something they collaborate on it, if they are not interested or stop being interested they don't collaborate and we have absolutely no way to ask people to work on stuff they are not interested to work on.

**Reply by greenhsu123** [reply]
The learn (Principles) team is slightly different though to the rest of the projects in the foundation as the project finished late last year, so we haven't actively worked on it for a while... how about we leave Principles & Patterns as a group for now (basically a Patterns group since we don't actively work on Principles as of now), and I agree that Markus will be a good lead (Sad that Bill can't re-join us)!

**Reply by camcash17** [reply]
are we expecting Principles work to pick up again soon? I suppose we have another meeting next week where it might be easier to discuss this.

**Reply by greenhsu123** [reply]
I don't think Principles work will reoccur until version 2, which I don't foresee until early next year! Another idea is to merge Principles with the project you are proposing @ camcash17, as these two projects are the most similar out of the 3. 

The only other work I foresee for Principles this year is the Earth Day event or any other promotional events, which the foundation's PMs and the Meetup group could take care of. 

Let's discuss this in more detail @ next week's meeting. :)
